From 3f2689dcea8946c83e0f050d7b7405e0e74ab96b Mon Sep 17 00:00:00 2001
From: Huiquan Zhong <huiquan.zhong@intel.com>
Date: Wed, 26 Aug 2015 23:24:23 +0800
Subject: [PATCH] HSU: fix UART TX DMA issue

Correct DMA address of dma_sync_single_for_device() for UART TX DMA.

Change-Id: Id5d817147b2076e4e16b713252291fb3ab1324a3
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-17465
Signed-off-by: Huiquan Zhong <huiquan.zhong@intel.com>
---
 drivers/tty/serial/hsu/hsu_dma.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/tty/serial/hsu/hsu_dma.c b/drivers/tty/serial/hsu/hsu_dma.c
index 31a79ef9c711..a2b2dec1ed2c 100644
--- a/drivers/tty/serial/hsu/hsu_dma.c
+++ b/drivers/tty/serial/hsu/hsu_dma.c
@@ -185,7 +185,7 @@ static void dw_dma_tx_done(void *arg)
 	struct hsu_dma_buffer *dbuf = &up->txbuf;
 	unsigned long flags;
 
-	dma_sync_single_for_cpu(dw_dma->txchan->device->dev, dbuf->dma_addr,
+	dma_sync_single_for_cpu(dw_dma->txchan->device->dev, dbuf->dma_addr + xmit->tail,
 				dbuf->dma_size, DMA_TO_DEVICE);
 
 	xmit->tail += dbuf->dma_size;
@@ -234,8 +234,8 @@ static void dw_dma_start_tx(struct uart_hsu_port *up)
 
 	dw_dma->tx_cookie = dmaengine_submit(desc);
 
-	dma_sync_single_for_device(dw_dma->txchan->device->dev, dbuf->dma_addr,
-				   dbuf->dma_size, DMA_TO_DEVICE);
+	dma_sync_single_for_device(dw_dma->txchan->device->dev, dbuf->dma_addr + xmit->tail,
+				dbuf->dma_size, DMA_TO_DEVICE);
 
 	dma_async_issue_pending(dw_dma->txchan);
 }
-- 
1.9.1

